include/inexor/vulkan-renderer/camera.hpp

```diff
@@ -1,127 +1,207 @@
 #pragma once
 
 #include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
 
-namespace inexor::vulkan_renderer {
-
-enum class CameraType { LOOKAT, FIRSTPERSON };
+#include <algorithm>
+#include <array>
+#include <vector>
 
-/// @brief A RAII wrapper class for camera position and movement.
-/// @todo Refactor method naming!
-class Camera {
-    /// @brief Update the view matrix.
-    /// @note The view matrix should only be updated if necessary.
-    void update_view_matrix();
+namespace inexor::vulkan_renderer {
 
-    CameraType m_type{CameraType::LOOKAT};
+enum class CameraMovement { FORWARD, BACKWARD, LEFT, RIGHT };
 
-    glm::vec3 m_position{};
-    glm::vec3 m_rotation{};
+// TODO: Implement more camera types.
+enum class CameraType { CAMERA_TYPE_LOOKAT };
 
-    float m_movement_speed{1.0f};
+/// @warning Not thread safe!
+class Camera {
+private:
+    /// The type of the camera. Currently only one type is implemented.
+    CameraType m_type{CameraType::CAMERA_TYPE_LOOKAT};
+    /// The start position of the camera.
+    glm::vec3 m_position{0.0f, 0.0f, 0.0f};
+    /// The vector of direction in which the camera is looking.
+    glm::vec3 m_front{0.0f, 0.0f, 1.0f};
+    /// The vector of direction which points to the right.
+    glm::vec3 m_right{1.0f, 0.0f, 0.0f};
+    /// The vector which indicates "upwards".
+    glm::vec3 m_up{0.0f, 1.0f, 0.0f};
```

Comment:
Just a short note:

We should put those vector. Somewhere more globally. They can be used inside the Octree code as well.

Reply:
Done. (Please review)